SonarQube.Client.Tests/Models/ServerSentEvents/SSEStreamTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarQube.Client.Models.ServerSentEvents;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarQube.Client.Tests.Models.ServerSentEvents
+{
+    [TestClass]
+    public class SSEStreamTests
+    {
+        [TestMethod]
+        public void BeginListening_CallsWriter()
+        {
+            var writerTask = Task.FromResult(5);
+            var writer = new Mock<ISSEStreamWriter>();
+            writer.Setup(x => x.BeginListening()).Returns(writerTask);
```

Comment:
ReturnsAsync

Reply:
Oof, it's part of Moq v4.5.28 and we use v4.14.5. I'll open a task to upgrade it.

Reply:
Are you going to change it now that you've updated the library in the other PR?

Reply:
I did in the other but I can't here since it's a void 🙈